snakemake/rules.py

```
@@ -261,6 +263,14 @@ def container_img(self):
     def container_img(self, container_img):
         self._container_img = container_img
 
+    @property
+    def sofware_env_spec(self):
+        return self._software_env_spec
+
+    @software_env_spec.setter
```

Ruff (0.8.2)

Undefined name `software_env_spec`

(F821)